src/algorithms/test/JsonToTypes.test.ts

```diff
@@ -0,0 +1,48 @@
+import { toEmpiricalData, toEmpiricalDatum } from '../types/JsonToTypes'
+import { EmpiricalData, EmpiricalDatum } from '../types/Param.types'
+import { CaseCounts, CaseCountsByCountry } from '../../assets/data/CaseCounts.types'
+import caseCountData from '../../assets/data/case_counts.json'
+
+describe('JsonToTypes', () => {
+  // This test exists mainly for the linter to detect type errors.
+  it('caseCount.json can be parsed into EmpiricalData type', () => {
+    const testCountry = 'United States of America'
+    const caseCountsByCountry: CaseCountsByCountry = caseCountData
+    const caseCounts: CaseCounts[] = caseCountsByCountry[testCountry]
+    const result: EmpiricalData = toEmpiricalData(caseCounts)
+    expect(result).toHaveLength(caseCountData[testCountry].length)
+  })
+
+  it('toEmpiricalData converts CaseCount "time" field to Date type.', () => {
+    const input: CaseCounts = {
+      time: '2020-03-17',
+      cases: 100,
+      deaths: 0,
+      hospitalized: 24,
+      ICU: 4,
+    }
+
+    const result: EmpiricalDatum = toEmpiricalDatum(input)
+    const expected = new Date('2020-03-17T00:00:00.000Z')
+
+    expect(result.time).toStrictEqual(expected)
+    expect(result.cases).toStrictEqual(100)
+    expect(result.deaths).toStrictEqual(0)
+    expect(result.hospitalized).toStrictEqual(24)
+    expect(result.ICU).toStrictEqual(4)
+  })
+
+  it('toEmpiricalData converts CaseCount null values to 0', () => {
+    const input: CaseCounts = {
+      time: '2020-03-17',
+      cases: 100,
+      deaths: null,
+      hospitalized: 24,
+      ICU: 4,
+    }
+
+    const result: EmpiricalDatum = toEmpiricalDatum(input)
+
+    expect(result.deaths).toStrictEqual(0)
+  })
```

Comment:
The `deaths`, `hospitalized` and `ICU` fields are sometimes null, and sometimes 0. The transformer should convert null value fields to 0.

Reply:
@abrie I am not sure that this is correct. We need to check our science super-heroes @nnoll @rneher 

This json comes from a data repo (meaning, from external sources) and definitely requires validation and sanitization, but I think we may want to handle missing data differently than just assigning a neutral value. For example, consider a real-world scenario: unknown number deaths is not the same as 0 deaths. It's a huge difference in the context of epidemic.

Alternatively, we may need to validate here and crash early to catch data errors. Again, different fields have different meaning and value (as in some are more important than others), so they may need different handling.

Reply:
Thanks for the close look. EmpiricalData type declares the deaths, hospitalized and ICU to be numbers; and the DeterministicLinePlot (which uses these values) checks only for 'truthiness', and never explicitly for a null. So converting to zero seemed like a reasonable assumption. But clearly there is a lot happening which cannot be inferred from the code – thank you for your explanation!

Reply:
My understanding is also that we use null in the JSON if no data is available. That being said, this should ideally only happen at the beginning of a data series until some data is available for that column (as the columns are cumulative). I don't think we check for this when preparing the JSON so far. Is it possible to start plotting certain values only from a later day onwards, where values are not null any more?

Reply:
@abrie Yep, so this is one of the cases where the type itself is incorrect, therefore validation could not be based on it.
Feel free to also check our data repo (in the submodule):
https://github.com/neherlab/covid19_scenarios_data
Your contributions have been very valuable here, so I am sure you could find something to hack there too :)

@noleti Yes, the missing values are currently not rendered, and they are not shown in the tooltip. This however would change if we would convert `null` to `0`.
In this particular case there is nothing wrong with `null` and I believe that data folks should send us `null` as it is. We just need to define what is the meaning of `null` and be prepared and handle it properly, both, in compile-time in typescript types and on runtime validation:
 - `null` but should be always preset? => data devs mistake  => crash early, before released
 - `null` and can be missing => OK =>  handle missing value during rendering